tiledb/sm/cpp_api/group_experimental.h

```diff
@@ -422,31 +423,23 @@ class Group {
   tiledb::Object member(uint64_t index) const {
     auto& ctx = ctx_.get();
     tiledb_ctx_t* c_ctx = ctx.ptr().get();
-    char* uri;
+    tiledb_string_t* uri;
     tiledb_object_t type;
-    char* name;
-    ctx.handle_error(tiledb_group_get_member_by_index(
+    tiledb_string_t* name;
+    ctx.handle_error(tiledb_group_get_member_by_index_v2(
         c_ctx, group_.get(), index, &uri, &type, &name));
-    std::string uri_str(uri);
-    std::free(uri);
-    std::optional<std::string> name_opt = std::nullopt;
-    if (name != nullptr) {
-      name_opt = name;
-      std::free(name);
-    }
-    return tiledb::Object(type, uri_str, name_opt);
+    return tiledb::Object(
+        type, impl::to_string(&uri).value(), impl::to_string(&name));
```

Comment:
Does the third argument of `Object` take an `optional`?

Reply:
Yes.